### PR TITLE
Primitive List feature(Part2: Nullability changes)

### PR DESCRIPTION
**Pull request by zaki50**

### CODE REVIEW

File: realm/realm-library/src/main/java/io/realm/RealmList.java [hunk 1]
```diff
@@ -380,7 +380,7 @@ public E remove(int location) {
      * @throws NullPointerException if {@code object} is {@code null}.
      */
     @Override
-    public boolean remove(Object object) {
+    public boolean remove(@Nullable Object object) {
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: Does this make sense? Can you remove the `null` element in a standard `ArrayList` ?

Comment (zaki50) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: Yes http://hg.openjdk.java.net/jdk7/jdk7/jdk/file/tip/src/share/classes/java/util/ArrayList.java#l473

Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: Huh, things I learned today 👍

**Comment by zaki50** [comment]
@cmelchior Added more `@Nullable` annotations.

**Comment by zaki50** [comment]
@cmelchior fixed all nullability issues in #5151

**Pull request merged**